src/main/scala/mesosphere/marathon/core/instance/InstanceStatus.scala

```diff
@@ -19,6 +19,11 @@ sealed trait InstanceStatus {
     }
   }
 
+  def isTerminal: Boolean = this match {
```

Comment:
Seems weird, and should be a val. Prefer it in the base trait and override for Terminal?